File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
const containsNearbyDuplicate = function(nums, k) {
    const seen = {};
    for (let i = 0; i < nums.length; i++) {
        const curr = nums[i];
        if (curr in seen && i - seen[curr] <= k) {
            return true;
        }
        seen[curr] = i;
    }
    return false;
};

File: easy/0219-contains-duplicate-ii/0219-contains-duplicate-ii.cs
public class Solution 
{
    public bool ContainsNearbyDuplicate(int[] nums, int k) 
    {
        IDictionary<int, int> seen = new Dictionary<int, int>();
        
        for (int i = 0; i < nums.Length; i++)
        {
            if (seen.ContainsKey(nums[i]))
            {
                if (i - seen[nums[i]] <= k)
                {
                    return true;
                }
                seen[nums[i]] = i;
            }
            else 
            {
                seen.Add(nums[i], i);
            }
        }
        
        return false;
    }
}